packages/lib/utils/chunkGenerator.ts

```diff
@@ -0,0 +1,52 @@
+import * as fs from 'fs';
+
+function readBytes(fd: number, sharedBuffer: Buffer) {
+	return new Promise<void>((resolve, reject) => {
+		fs.read(fd, sharedBuffer, 0, sharedBuffer.length, null, (error) => {
+			if (error) {
+				return reject(error);
+			}
+			resolve();
+		});
+	});
+}
+
+async function* generateChunks(
+	filePath: string,
+	size: number
+): AsyncGenerator<{ index: number, chunksNo: number, chunk: Buffer }> {
+	const sharedBuffer = Buffer.alloc(size);
+	let stats;
+	let fd;
+	let bytesRead = 0;
+	let end = size;
+
+	try {
+		stats = fs.statSync(filePath);
+		fd = fs.openSync(filePath, 'r');
+	} catch (_) {
+		throw new Error('Failed to open file');
+	}
+
+	const chunksNo = Math.ceil(stats.size / size);
+	for (let i = 0; i < chunksNo; i++) {
+		try {
+			await readBytes(fd, sharedBuffer);
+		} catch (_) {
+			throw new Error('Failed to read bytes from file');
```

Comment:
worth showing the original error to the user?

Reply:
it helped me in debugging but was very ugly, I think we shouldn't show it.  what do you think?

Reply:
An error being ugly is lot less of a problem than an error being silenced, so please show the full error using the above pattern.